src/InboxMonitor.vala

```diff
@@ -0,0 +1,238 @@
+            return;
+        }
+
+        var inbox_folder = inbox_folders.get (source);
+        if (inbox_folder != null) {
+            debug ("[%s] Refreshing…", source.display_name);
+
+            try {
+                inbox_folder.refresh_info_sync (null);
+
+            } catch (Error e) {
+                warning ("[%s] Error refreshing: %s", source.display_name, e.message);
+            }
+        }
+    }
+
+    private void inbox_folder_changed (E.Source source, Camel.FolderChangeInfo changes) {
+        var inbox_folder = inbox_folders.get (source);
+        if (inbox_folder == null) {
+            return;
+        }
+
+        unowned var added_uids = changes.get_added_uids ();
+        if (added_uids != null) {
+            var sender_names = new GenericSet<string> (str_hash, str_equal);
+            var unseen_message_infos = new SList<Camel.MessageInfo> ();
+
+            added_uids.foreach ((added_uid) => {
+                var message_info = inbox_folder.get_message_info (added_uid);
+
+                if (!(Camel.MessageFlags.SEEN in message_info.flags)) {
+                    unowned string? sender_address;
+                    unowned string? sender_name;
+
+                    var camel_address = new Camel.InternetAddress ();
+                    camel_address.unformat (message_info.from);
+                    camel_address.get (0, out sender_name, out sender_address);
+
+                    if (sender_name == null) {
+                        sender_name = sender_address;
+                    }
+
+                    sender_names.add (sender_name);
+                    unseen_message_infos.append (message_info);
+                }
+            });
+
+            var unseen_message_infos_length = unseen_message_infos.length ();
+            if (unseen_message_infos_length == 1) {
+                var unseen_message_info = unseen_message_infos.nth_data (0);
+
+                var notification = new GLib.Notification (_("%s - %s").printf (sender_names.iterator ().next_value (), inbox_folder.parent_store.display_name));
+                notification.set_body (unseen_message_info.subject);
+                GLib.Application.get_default ().send_notification (unseen_message_info.uid, notification);
+
+            } else if (unseen_message_infos_length > 1) {
+                GLib.Notification notification;
+                string messages_count = ngettext ("%u new message", "%u new messages", unseen_message_infos_length).printf (unseen_message_infos_length);
+
+                if (sender_names.length == 1) {
+                    var sender_name = sender_names.iterator ().next_value ();
+
+                    notification = new GLib.Notification (_("%s - %s").printf (sender_name, inbox_folder.parent_store.display_name));
+                    ///TRANSLATORS: The %s represents the number of new messages translated in your language, e.g. "2 new messages"
+                    notification.set_body (_("%s").printf (messages_count));
```

Comment:
Nothing left to translate here right? :)
```suggestion
                    notification.set_body (messages_count);
```

Reply:
Ok - wasn't 100% sure if ngettext does everything for us. But this is obviously the case so changed this accordingly.